### PR TITLE
Argwhere and flatnonzero

### PR DESCRIPTION
**Pull request by mfoerste4**
This PR adds functionality for `np.argwhere `and `np.flatnonzero` to cuNumeric.

### np.argwhere
Implemented similar to `np.nonzero` task, but with different return shape

### np.flatnonzero
Just forwarded to `np.nonzero(np.ravel(a))[0]`

### CODE REVIEW

**Comment by mfoerste4** [comment]
@magnatelee I suggested you for review, feel free to forward to someone else.

File: src/cunumeric.mk [hunk 1]
```diff
@@ -50,6 +50,7 @@ GEN_CPU_SRC += cunumeric/ternary/where.cc               \
 							 cunumeric/matrix/trsm.cc                 \
 							 cunumeric/matrix/util.cc                 \
 							 cunumeric/random/rand.cc                 \
+							 cunumeric/search/argwhere.cc              \
```
Comment (magnatelee) [inline src/cunumeric.mk:53]: a nitpick: any reason that you don't align the backslash with the rest?

File: src/cunumeric/search/argwhere.cc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      auto in_p = pitches.unflatten(idx, rect.lo);
+
+      if (input[in_p] != VAL(0)) {
+        for (int i = 0; i < DIM; ++i) { out[Point<2>(out_idx, i)] = in_p[i]; }
+        out_idx++;
+      }
+    }
+
+    assert(size == out_idx);
+    // out_array.return_data(out, Point<2>(size, DIM));
```
Comment (magnatelee) [inline src/cunumeric/search/argwhere.cc:54]: please remove this line

**Review by magnatelee — CHANGES_REQUESTED** [review]

**Comment by magnatelee** [comment]
left two petty comments. otherwise looks good to me. great to see that you reuse the kernels from the `nonzero` task!

File: src/cunumeric.mk (hunk 1 above)
Comment (mfoerste4) [reply src/cunumeric.mk:53]: Nope, just an accident. I will adjust the alignment.

**Comment by mfoerste4** [comment]
> left two petty comments. otherwise looks good to me. great to see that you reuse the kernels from the `nonzero` task!
@magnatelee , please verify requested changes to unblock merge.

**Review by magnatelee — APPROVED** [review]

**Comment by magnatelee** [comment]
@mfoerste4 approved. feel free to merge it whenever you can.

**Pull request merged**